imagegw/shifter_imagegw/dockerv2.py

```diff
@@ -89,21 +89,25 @@ def _setup_http_conn(url, cacert=None):
     """Prepare http connection object and return it."""
     (protocol, url) = url.split('://', 1)
     conn = None
-    port = 443
+    if protocol == 'http':
+        port=80
+    else:
+        port = 443
     if url.find('/') >= 0:
         (server, _) = url.split('/', 1)
     else:
         server = url
     if ':' in server:
-        (server, port) = server.split(':')
+        (server, ports) = server.split(':')
```

Comment:
it might be safer to limit the split to just one split, e.g.,
`(server,port) = server.split(':', 1)`